src/applications/personalization/dashboard-2/components/Dashboard.jsx

```diff
@@ -2,73 +2,153 @@ import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
 import isEmpty from 'lodash/isEmpty';
 
+import '../sass/dashboard.scss';
+
 import Breadcrumbs from '@department-of-veterans-affairs/component-library/Breadcrumbs';
+import LoadingIndicator from '@department-of-veterans-affairs/component-library/LoadingIndicator';
+
 import { focusElement } from '~/platform/utilities/ui';
+import {
+  isLOA3 as isLOA3Selector,
+  isLOA1 as isLOA1Selector,
+  isLoggedIn as isLoggedInSelector,
+} from '~/platform/user/selectors';
+
+import NameTag from '~/applications/personalization/profile/components/NameTag';
+import { fetchTotalDisabilityRating as fetchTotalDisabilityRatingAction } from '~/applications/personalization/rated-disabilities/actions';
 
 import {
   fetchMilitaryInformation as fetchMilitaryInformationAction,
   fetchHero as fetchHeroAction,
-  fetchPersonalInformation as fetchPersonalInformationAction,
 } from '@@profile/actions';
 
-import { fetchTotalDisabilityRating as fetchTotalDisabilityRatingAction } from '~/applications/personalization/rated-disabilities/actions';
-
-import NameTag from '~/applications/personalization/profile/components/NameTag';
-
+import IdentityNotVerified from './IdentityNotVerified';
 import ApplyForBenefits from './apply-for-benefits/ApplyForBenefits';
 import ClaimsAndAppeals from './claims-and-appeals/ClaimsAndAppeals';
 import HealthCare from './health-care/HealthCare';
 
-const Dashboard = props => {
-  useEffect(() => {
-    props.fetchFullName();
-    props.fetchPersonalInformation();
-    props.fetchMilitaryInformation();
-    props.fetchTotalDisabilityRating();
-    focusElement('#dashboard-title');
-  }, []);
-
-  return (
-    <>
-      {isEmpty(props.hero?.errors) && (
-        <NameTag
-          showUpdatedNameTag
-          totalDisabilityRating={props.totalDisabilityRating}
-        />
-      )}
-      <div className="vads-l-grid-container vads-u-padding-x--0">
-        <Breadcrumbs>
-          <a href="/" key="home">
-            Home
-          </a>
-          <span className="vads-u-color--black" key="dashboard">
-            <strong>My VA</strong>
-          </span>
-        </Breadcrumbs>
-
-        <h1 id="dashboard-title" data-testid="dashboard-title" tabIndex="-1">
-          My VA
-        </h1>
-
-        <ClaimsAndAppeals />
-        <HealthCare />
-        <ApplyForBenefits />
-      </div>
-    </>
+// content to show if the component is waiting for data to load. This loader
+// matches the loader shown by the RequiredLoginView component, so when the
+// RequiredLoginView is done with its loading and this component takes over, it
+// appears seamless to the user.
+const Loader = () => (
+  <div className="vads-u-margin-y--5">
+    <LoadingIndicator setFocus message="Loading your information..." />
+  </div>
+);
+
+const Dashboard = ({
+  fetchFullName,
+  fetchMilitaryInformation,
+  fetchTotalDisabilityRating,
+  isLOA3,
+  showLoader,
+  ...props
```

Comment:
Can we get rid of the props spreading here? 😄

Reply:
We can, but let me explain my reasoning for this 😀 

I am only pulling off the props that are needed by the `useEffect` hooks so I can pass them to the dependency arrays. And since the `mapStateToProps` was in flux, it's nice to not have to keep the destructuring up to date. In other words it's nice to just pull off the props that really need to be their own variables and then access the rest off of the `props` object the remaining props are collected into. It's totally a matter of opinion, but I kind of like how `props.showValidateIdentityAlert` looks when we are creating the JSX. It makes it clear where `showValidateIdentityAlert` comes from.